File: frontend/src/data/proceedingType.ts
import { ProceedingType, ProceedingTypeLabels } from "@/domain/objectValue"

export const proceedingType = [
  {
    label: ProceedingTypeLabels.VERFASSUNGSBESCHWERDE,
    value: ProceedingType.VERFASSUNGSBESCHWERDE,
  },
  {
    label: ProceedingTypeLabels.ORGANSTREITVERFAHREN,
    value: ProceedingType.ORGANSTREITVERFAHREN,
  },
  {
    label: ProceedingTypeLabels.KONKRETE_NORMENKONTROLLE,
    value: ProceedingType.KONKRETE_NORMENKONTROLLE,
  },
  {
    label: ProceedingTypeLabels.ABSTRAKTE_NORMENKONTROLLE,
    value: ProceedingType.ABSTRAKTE_NORMENKONTROLLE,
  },
  {
    label: ProceedingTypeLabels.BUND_LAENDER_STREITIGKEIT,
    value: ProceedingType.BUND_LAENDER_STREITIGKEIT,
  },
  {
    label: ProceedingTypeLabels.WAHLPRUEFUNGSVERFAHREN,
    value: ProceedingType.WAHLPRUEFUNGSVERFAHREN,
  },
  {
    label: ProceedingTypeLabels.EINSTWEILIGER_RECHTSSCHUTZ,
    value: ProceedingType.EINSTWEILIGER_RECHTSSCHUTZ,
  },
  {
    label: ProceedingTypeLabels.SONSTIGE_VERFAHREN,
    value: ProceedingType.SONSTIGE_VERFAHREN,
  },
]

File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/ProceedingType.java
package de.bund.digitalservice.ris.caselaw.domain;

public enum ProceedingType {
  VERFASSUNGSBESCHWERDE,
  ORGANSTREITVERFAHREN,
  KONKRETE_NORMENKONTROLLE,
  ABSTRAKTE_NORMENKONTROLLE,
  BUND_LAENDER_STREITIGKEIT,
  WAHLPRUEFUNGSVERFAHREN,
  EINSTWEILIGER_RECHTSSCHUTZ,
  SONSTIGE_VERFAHREN;

  @Override
  public String toString() {
    return switch (this) {
      case VERFASSUNGSBESCHWERDE -> "Verfassungsbeschwerde";
      case ORGANSTREITVERFAHREN -> "Organstreitverfahren";
      case KONKRETE_NORMENKONTROLLE -> "Konkrete Normenkontrolle";
      case ABSTRAKTE_NORMENKONTROLLE -> "Abstrakte Normenkontrolle";
      case BUND_LAENDER_STREITIGKEIT -> "Bund-Länder-Streitigkeit";
      case WAHLPRUEFUNGSVERFAHREN -> "Wahlprüfungsverfahren";
      case EINSTWEILIGER_RECHTSSCHUTZ -> "Einstweiliger Rechtsschutz";
      case SONSTIGE_VERFAHREN -> "Sonstige Verfahren";
    };
  }
}
